packages/node_modules/@ciscospark/plugin-messages/src/messages.js

```diff
@@ -193,6 +209,33 @@ const Messages = SparkPlugin.extend({
       .then((res) => res.body.items || res.body);
   },
 
+  /**
+   * Marks a message as read.
+   * @instance
+   * @memberof Messages
+   * @param {MessageObject} message
+   * @returns {Promise<MessageObject>}
+   */
+  markAsRead(message) {
+    const activity = {
+      id: deconstructHydraId(message.id).id
+    };
+    const conversation = {
+      id: deconstructHydraId(message.roomId).id
+    };
+
+    return this.spark.internal.conversation.acknowledge(conversation, activity)
+      .then((ack) => {
```

Comment:
this can become
```javascript
.then((ack) => ({
  id: constructHydraId(ack.id),
  messageId: constructHydraId(hydraTypes.MESSAGE, ack.object.id),
  markedAsRead: ack.published
}))
```